File: LeetCode/Medium/0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        
        int l = 0;
        int r = people.length - 1;
        int ret = 0;
        
        while (r >= l) {
            if (people[l] + people[r] <= limit) ++l;
            r--;
            ++ret;
        }
        
        return ret;
    }
}

File: LeetCode/Medium/0881-boats-to-save-people/0881-boats-to-save-people.go
func numRescueBoats(people []int, limit int) int {
    sort.Ints(people)
        
    l := 0
    r := len(people) - 1
    ret := 0
    
    for r >= l {
        if people[l] + people[r] <= limit { l++ }
        r--
        ret++
    }

    return ret
}